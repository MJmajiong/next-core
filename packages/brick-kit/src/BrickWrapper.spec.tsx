import React from "react";
import { mount, shallow } from "enzyme";
import { BrickWrapper } from "./BrickWrapper";
import { Empty, Table } from "antd";
import { getRuntime } from "./runtime";

jest.mock("i18next", () => ({ language: "zh" }));
jest.mock("antd/es/locale/zh_CN", () => "zh_CN");
jest.mock("antd/es/locale/en_US", () => "en_US");
jest.mock("./runtime");

(getRuntime as jest.Mock).mockImplementation(() => ({
  getFeatureFlags: () => ({
    "enable-feature": true,
    "enable-say-hi": false,
    "enable-say-hello": true,
  }),
}));

describe("brick wrapper", () => {
  it("should work", () => {
    const wrapper = shallow(
      <BrickWrapper>
        <div>hello, brick-wrapper</div>
      </BrickWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should work", () => {
    const wrapper = mount(
      <BrickWrapper>
        <Table />
      </BrickWrapper>
    );
    expect(wrapper.find(Empty).length).toBe(1);
  });
});
