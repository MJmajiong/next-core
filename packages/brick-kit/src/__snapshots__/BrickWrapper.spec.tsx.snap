// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brick wrapper should work 1`] = `
<LegacyErrorBoundary
  i18n={
    Object {
      "language": "zh-CN",
    }
  }
  t={[Function]}
>
  <ContextProvider
    value={Object {}}
  >
    <ConfigProvider
      autoInsertSpaceInButton={false}
      locale="zh_CN"
      renderEmpty={[Function]}
    >
      <div>
        hello, brick-wrapper
      </div>
    </ConfigProvider>
  </ContextProvider>
</LegacyErrorBoundary>
`;
